plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/server/debug/OpenShiftDebugUtils.java

```diff
@@ -0,0 +1,260 @@
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.core.server.debug;
+
+import static org.eclipse.jdt.launching.IJavaLaunchConfigurationConstants.ID_REMOTE_JAVA_APPLICATION;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang.math.NumberUtils;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.core.runtime.jobs.IJobChangeEvent;
+import org.eclipse.core.runtime.jobs.JobChangeAdapter;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunch;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchConfigurationType;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.debug.core.ILaunchManager;
+import org.eclipse.jdt.launching.IJavaLaunchConfigurationConstants;
+import org.eclipse.wst.server.core.IServer;
+import org.jboss.tools.openshift.common.core.OpenShiftCoreException;
+import org.jboss.tools.openshift.common.core.connection.ConnectionsRegistrySingleton;
+import org.jboss.tools.openshift.internal.core.models.PortSpecAdapter;
+import org.jboss.tools.openshift.internal.core.util.ResourceUtils;
+
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.capability.CapabilityVisitor;
+import com.openshift.restclient.capability.resources.IClientCapability;
+import com.openshift.restclient.model.IContainer;
+import com.openshift.restclient.model.IDeploymentConfig;
+import com.openshift.restclient.model.IEnvironmentVariable;
+import com.openshift.restclient.model.IPod;
+import com.openshift.restclient.model.IPort;
+
+public class OpenShiftDebugUtils {
+
+	private ILaunchManager launchManager;
+
+	public static OpenShiftDebugUtils get() {
+		return get(DebugPlugin.getDefault().getLaunchManager());
+	}
+	
+	/** For testing purposes **/
+	public static OpenShiftDebugUtils get(ILaunchManager launchManager) {
+		return new OpenShiftDebugUtils(launchManager);
+	}
+	
+	private OpenShiftDebugUtils(ILaunchManager launchManager) {
+		this.launchManager = launchManager;
+	}
+	
+	public DebuggingContext enableDebugMode(IDeploymentConfig deploymentConfig, DebuggingContext debugContext, IProgressMonitor monitor) throws CoreException {
+		assert deploymentConfig != null;
```

Comment:
Shouldn't we avoid the `asset` keyword ? Are we sure that Eclipse is started with the `-ea` switch ?